src/pages/permissions/CreateTLSIdentity.tsx

```diff
@@ -0,0 +1,36 @@
+import { useState, type FC } from "react";
+import CreateIdentityModal from "./CreateIdentityModal";
+import CreateTLSIdentityPanel from "./panels/CreateTLSIdentityPanel";
+import { usePortal } from "@canonical/react-components";
+import usePanelParams, { panels } from "util/usePanelParams";
+
+const CreateTLSIdentity: FC = () => {
+  const panelParams = usePanelParams();
+  const { openPortal, closePortal, isOpen, Portal } = usePortal();
+  const [token, setToken] = useState("");
+  const [identityName, setIdentityName] = useState("");
+
+  return (
+    <>
+      {isOpen && (
+        <Portal>
+          <CreateIdentityModal
+            onClose={closePortal}
+            token={token}
+            identityName={identityName}
+          />
+        </Portal>
+      )}
+
+      {panelParams.panel === panels.createTLSIdentity && (
+        <CreateTLSIdentityPanel
+          openPortal={openPortal}
+          onSuccess={setToken}
+          setIdentityName={setIdentityName}
```

Comment:
```suggestion
          openPortal={openPortal}
          onSuccess={(token, identityName) => { setToken ... setIdentityName ... openPortal}}
```
I think the interface can be much smaller and more concise, as all the updates relate to a single event.